integrationtests/gitjob/git/git_test.go

```diff
@@ -62,309 +65,373 @@ var (
+	tmpDir, err := os.MkdirTemp("", "gogs")
+	Expect(err).NotTo(HaveOccurred())
+
+	container, url, err = createGogsContainer(ctx, tmpDir)
+	Expect(err).NotTo(HaveOccurred())
+})
+
+var _ = AfterSuite(func() {
+	if container != nil {
+		ctx := context.Background()
+		err := container.Terminate(ctx)
+		Expect(err).NotTo(HaveOccurred())
+	}
+})
+
+var _ = Describe("Git Fetch", func() {
+	Describe("LatestCommit NoAuth", func() {
+		var ctx context.Context
+		var ctlr *gomock.Controller
+
+		BeforeEach(func() {
+			ctx = context.Background()
+			ctlr = gomock.NewController(GinkgoT())
+			config.Set(&config.Config{})
+		})
 
-func TestLatestCommit_NoAuth(t *testing.T) {
-	ctlr := gomock.NewController(t)
-	defer ctlr.Finish()
-	ctx := context.Background()
+		AfterEach(func() {
+			ctlr.Finish()
+		})
 
-	tests := map[string]struct {
-		gitrepo        *v1alpha1.GitRepo
-		expectedCommit string
-		expectedErr    error
-	}{
-		"public repo": {
-			gitrepo: &v1alpha1.GitRepo{
+		It("public repo", func() {
+			gitrepo := &v1alpha1.GitRepo{
 				Spec: v1alpha1.GitRepoSpec{
 					Repo:   url + "/test/public-repo",
 					Branch: "master",
 				},
-			},
-			expectedCommit: latestCommitPublicRepo,
-			expectedErr:    nil,
-		},
-		"private repo": {
-			gitrepo: &v1alpha1.GitRepo{
+			}
+
+			f := fleetgit.Fetch{}
+			client := mocks.NewMockK8sClient(ctlr)
+			// May be called multiple times, including calls to get Rancher CA bundle secrets
+			client.EXPECT().Get(ctx, gomock.Any(), gomock.Any()).
+				Return(apierrors.NewNotFound(schema.GroupResource{}, "notfound")).AnyTimes()
+			latestCommit, err := f.LatestCommit(ctx, gitrepo, client)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(latestCommit).To(Equal(latestCommitPublicRepo))
+		})
+
+		It("private repo", func() {
+			gitrepo := &v1alpha1.GitRepo{
 				Spec: v1alpha1.GitRepoSpec{
 					Repo:   url + "/test/private-repo",
 					Branch: "master",
 				},
-			},
-			expectedCommit: "",
-			expectedErr:    transport.ErrAuthenticationRequired,
-		},
-	}
+			}
 
-	config.Set(&config.Config{})
-	for name, test := range tests {
-		t.Run(name, func(t *testing.T) {
-			f := git.Fetch{}
+			f := fleetgit.Fetch{}
 			client := mocks.NewMockK8sClient(ctlr)
 			// May be called multiple times, including calls to get Rancher CA bundle secrets
 			client.EXPECT().Get(ctx, gomock.Any(), gomock.Any()).
 				Return(apierrors.NewNotFound(schema.GroupResource{}, "notfound")).AnyTimes()
-			latestCommit, err := f.LatestCommit(ctx, test.gitrepo, client)
-			if test.expectedErr == nil {
-				require.NoError(t, err)
-			} else {
-				require.Contains(t, err.Error(), test.expectedErr.Error())
-			}
-			if latestCommit != test.expectedCommit {
-				t.Errorf("latestCommit doesn't match. got %s, expected %s", latestCommit, test.expectedCommit)
-			}
+			latestCommit, err := f.LatestCommit(ctx, gitrepo, client)
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(ContainSubstring(transport.ErrAuthenticationRequired.Error()))
```

Comment:
This would be more idiomatic:
```suggestion
			Expect(err).To(MatchError(transport.ErrAuthenticationRequired)) 
```